src/RankingManagerC.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+#include "EventComponent.h"
+#include "Factory.h"
+
+#include <string>
+#include <vector>
+
+DECLARE_FACTORY(RankingManagerC);
+
+class SpawnerEnemiesEC;
+class SpawnerEC;
+
+struct RankingPosition {
+    int round;
+    std::string name;
+
+    bool operator<(RankingPosition const& b) { return round > b.round; }
+};
```

Comment:
Just an idea, what if you define `friend`s for `std::ostream`? That way you'd be able to do:

```cpp
friend std::ostream& operator<<(std::ostream& out, RankingPosition& position) {
  out << position.round << ' ' << position.name << '\n';
  return out;
}
```

```cpp
std::cout << rankingPosition << '\n';
```

And that'd write the name (`-` if not set) and the points, likewise:

```cpp
std::cin >> rankingPosition;
```

Which would internally do:

```cpp
friend std::ostream& operator>>(std::ostream& in, RankingPosition& position) {
  in >> position.round >> position.name;
  return in;
}
```

More information: https://stackoverflow.com/questions/38093045/what-does-friend-stdostream-operatorstdostream-out-linkedlist-list

I think you can make `RankingManagerC` much tidier by leaving the serialization and deserialization process to the struct itself, following OOP's best practices about encapsulation.

Reply:
C2676	'<<' binario: 'std::ostream' no define este operador o una conversión a un tipo aceptable para el operador predefinido